app/models/plugins/fundraiser.rb

```diff
@@ -25,7 +25,9 @@ class Plugins::Fundraiser < ApplicationRecord
   belongs_to :page, touch: true
   belongs_to :donation_band
 
-  after_create :create_donations_thermometer
+  # After creating a fundraiser plugin, also create a donations thermometer, unless one already exists for the page.
+  after_create :create_donations_thermometer,
+               unless: proc { |plugin| Plugins::DonationsThermometer.where(page_id: plugin.page.id).exists? }
 
```

Comment:
Does this mean that, when cloning, we are also cloning the thermometer plugin plus its settings?

Reply:
Yes, we `.dup` it like the other plugins. This is where the issue was (in PageCloner): 
```  def plugins
    cloned_page.plugins.each(&:destroy!)
    page.plugins.each do |plugin|
      plugin.dup.tap do |clone|
        clone.page = cloned_page
        update_form(clone) if @override_forms
        clone.save!
      end
    end
  end```
If the page had a fundraiser and donations thermometer plugin (those go hand in hand sot it would have had both) - it'd clone the donations thermometer plugin as intended, but when it'd called `.dup` on the fundraiser plugin, it'd create a new fundraiser plugin as well as a donations thermometer plugin before updating the association on the new fundraiser plugin to the cloned page. The donations thermometer plugin created like this would stay and point to the original page - so the original page would always get a new donations thermometer when it'd get cloned.

The solution was to add a conditional to the after_create AR callback on the Fundraiser to only create the DonationsThermometer if one didn't already exist for its page. I took the liberty of doing this because there is already logic elsewhere (page plugin switcher) that'll only keep one of the thermometer plugins if there were more than one on a page.